File: src/main/java/com/condominium/management/charge/entity/Charge.java
package com.condominium.management.charge.entity;

import com.condominium.management.unit.entity.Unit;
import jakarta.persistence.*;
import lombok.*;

import java.math.BigDecimal;
import java.time.LocalDate;
import java.time.LocalDateTime;

@Entity
@Table(name = "charges")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Charge {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String description;

    @Column(nullable = false, precision = 12, scale = 2)
    private BigDecimal amount;

    @Column(nullable = false)
    private LocalDate dueDate;

    @Column(nullable = false, length = 7)
    private String referenceMonth;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private ChargeStatus status;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "unit_id", nullable = false)
    private Unit unit;

    @Column(nullable = false)
    private LocalDateTime createdAt;
}

File: src/types/charge.ts
export type ChargeStatus =
    | "PENDING"
    | "PAID"
    | "OVERDUE"
    | "CANCELLED";

export interface Charge {
    id: number;
    description: string;
    amount: number;
    dueDate: string;
    referenceMonth: string;
    status: ChargeStatus;
    unitId: number;
    unitNumber: string;
}

export interface ChargeRequest {
    description: string;
    amount: number;
    dueDate: string;
    referenceMonth: string;
    unitId: number;
}